avocado/core/future/settings.py

```diff
@@ -0,0 +1,516 @@
+                 parser=None, short_arg=None, long_arg=None,
+                 positional_arg=False, choices=None, nargs=None,
+                 metavar=None, required=None, action=None):
+        self.namespace = namespace
+        self.help_msg = help_msg
+        self.key_type = key_type
+        self.default = default
+        self.parser = parser
+        self.short_arg = short_arg
+        self.long_arg = long_arg
+        self.positional_arg = positional_arg
+        self.choices = choices
+        self.nargs = nargs
+        self._metavar = metavar
+        self.required = required
+        self._action = action
+        self._value = None
+
+    @property
+    def action(self):
+        if self.key_type is bool:
+            # action is automatic when using bool types
+            if self.default is False:
+                return 'store_true'
+            else:
+                return 'store_false'
+        return self._action
+
+    @property
+    def section(self):
+        return '.'.join(self.namespace.split('.')[:-1])
+
+    @property
+    def key(self):
+        return self.namespace.split('.')[-1]
+
+    @property
+    def value(self):
+        if self._value:
+            return self._value
+        return self.default
+
+    @property
+    def name_or_tags(self):
+        if self.positional_arg:
+            return self.key
+        tags = []
+        if self.short_arg:
+            tags.append(self.short_arg)
+        if self.long_arg:
+            tags.append(self.long_arg)
+        return tags
+
+    @property
+    def argparse_type(self):
+        # type is automatic when using lists Argparse do not support list
+        # on command-line. They are comma separated strings that will be
+        # converted to list here.
+        if self.key_type is list:
+            return str
+        else:
+            return self.key_type
+
+    @property
+    def metavar(self):
+        if self.positional_arg:
+            if self._metavar is None:
+                return self.key
+        return self._metavar
+
+    @property
+    def arg_parse_args(self):
+        args = {'help': self.help_msg,
+                'default': None}
+
+        if self.nargs:
+            args['nargs'] = self.nargs
+        if self.metavar:
+            args['metavar'] = self.metavar
+        if self.choices:
+            args['choices'] = self.choices
+        if self.action:
+            args['action'] = self.action
+
+        if self.key_type is not bool:
+            # We don't specify type for bool
+            args['type'] = self.argparse_type
+
+        if not self.positional_arg:
+            args['required'] = self.required
+            args['dest'] = self.namespace  # most of the magic is here
+
+        return args
+
+    def _as_list(self, value):
+        if value == '':
+            return []
+
+        if isinstance(value, str):
+            return value.strip('[]').split(',')
```

Comment:
I think the behavior here should be a bit more uniform. What I mean is, I believe these should have the result output:

1. `''`
2. `[]`
3. `[,,,]`

IMO, all of those should return an empty list.  Also, I think it would be nice to not return `None` implicitly if a value that can not be turned into a list. This would require something like the following:

```diff
diff --git a/avocado/core/future/settings.py b/avocado/core/future/settings.py
index 7ae7ed57..8cc562ba 100644
--- a/avocado/core/future/settings.py
+++ b/avocado/core/future/settings.py
@@ -171,15 +171,14 @@ class ConfigOption:
         return args
 
     def _as_list(self, value):
-        if value == '':
-            return []
-
         if isinstance(value, str):
-            return value.strip('[]').split(',')
+            return [v for v in value.strip('[]').split(',') if v]
 
         if isinstance(value, list):
             return value
 
+        raise ValueError(f'"{value}" could not be convrted into a list')
+
     def set_value(self, value, convert=False):
         dst_type = self.key_type
```

Reply:
BTW, feel free to use this: https://github.com/clebergnu/avocado/commit/f1d77a9411125b8c9957cbf36bc9d943ea4f0b65

Reply:
Sure, yes, I agree. Thanks. Fixed in v2.